pkg/client/unversioned/conditions.go

```diff
@@ -17,11 +17,69 @@ limitations under the License.
 package unversioned
 
 import (
+	"time"
+
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/apis/extensions"
 	"k8s.io/kubernetes/pkg/util/wait"
 )
 
+// DefaultRetry is the recommended retry for a conflict where multiple clients
+// are making changes to the same resource.
+var DefaultRetry = wait.Backoff{
+	Steps:    5,
+	Duration: 10 * time.Millisecond,
+	Factor:   1.0,
+	Jitter:   0.1,
+}
+
+// DefaultBackoff is the recommended backoff for a conflict where a client
+// may be attempting to make an unrelated modification to a resource under
+// active management by one or more controllers.
+var DefaultBackoff = wait.Backoff{
+	Steps:    4,
+	Duration: 10 * time.Millisecond,
+	Factor:   5.0,
```

Comment:
In practice a lot of controllers do a flurry of updates right after a
resource is created.  We see this on deployments, builds, pods, replication
controllers, and PVCs.  And service accounts.

On Oct 28, 2015, at 7:16 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/client/unversioned/conditions.go
<https://github.com/kubernetes/kubernetes/pull/16067#discussion_r43331509>:

> +// DefaultRetry is the recommended retry for a conflict where multiple clients
> +// are making changes to the same resource.
> +var DefaultRetry = wait.Backoff{
> +	Steps:    5,
> +	Duration: 10 * time.Millisecond,
> +	Factor:   1.0,
> +	Jitter:   0.1,
> +}
> +
> +// DefaultBackoff is the recommended backoff for a conflict where a client
> +// may be attempting to make an unrelated modification to a resource under
> +// active management by one or more controllers.
> +var DefaultBackoff = wait.Backoff{
> +	Steps:    4,
> +	Duration: 10 * time.Millisecond,
> +	Factor:   5.0,

The assumption that would make this make sense is that a controller that
just made a change is less likely to make another change in the more
distant future, but I don't see why that assumption should be true.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/16067/files#r43331509>.